packages/react/src/components/heliopay-container/index.tsx

```diff
@@ -79,26 +91,55 @@ const HelioPayContainer: FC<HeliopayContainerProps> = ({
     null
   );
 
+  const [actualPrice, setActualPrice] = useState(0);
+
   useEffect(() => {
     initCluster(cluster);
   }, [cluster]);
 
-  useEffect(() => {
-    if (
-      paymentDetails?.features?.requireDeliveryAddress &&
-      paymentDetails?.features?.requireCountry
-    ) {
-      getCountry();
-    }
-  }, [paymentDetails]);
-
   const generateAllowedCurrencies = () => {
     const allowedCurrenciesTemp = currencyList.filter((currency) =>
       supportedCurrencies?.includes(currency.symbol)
     );
     setAllowedCurrencies(allowedCurrenciesTemp);
   };
 
+  const delay = (cb: () => void, ms: number) => {
+    // eslint-disable-next-line no-new
+    new Promise((Resolve) => {
+      setTimeout(() => {
+        Resolve(cb());
+      }, ms);
+    });
+  };
+
+  const nowMS = now();
+  const expirationMS = tokenExpiration ? tokenExpiration * 1000 : 0;
+
+  useEffect(() => {
+    if (
+      tokenExpiration &&
+      paymentDetails?.fixedCurrency?.price &&
+      paymentDetails?.currency &&
+      paymentDetails?.fixedCurrency?.price &&
+      expirationMS > nowMS
+    ) {
+      delay(() => {
+        getTokenPrice({
+          from: paymentDetails?.fixedCurrency?.currency,
+          to: paymentDetails?.currency?.symbol,
+          amount: paymentDetails?.fixedCurrency?.price,
+        });
+      }, expirationMS - nowMS);
+    }
+  }, [
+    tokenExpiration,
+    paymentDetails?.fixedCurrency,
+    paymentDetails?.currency,
+    paymentDetails?.fixedCurrency,
+    expirationMS,
```

Comment:
Maybe am wrong here but just make sure paymentDetails is coming from HelioPay props not paymentRequest details